src/app/api/tasks/templates/public/public.serializer.ts

```diff
@@ -0,0 +1,27 @@
+import { TemplateResponsePublic, TemplateResponsePublicSchema } from '@/app/api/tasks/templates/public/public.dto'
+import { toRFC3339 } from '@/utils/dateHelper'
+import { TaskTemplate } from '@prisma/client'
+
+export class PublicTemplateSerializer {
+  static serialize(template: TaskTemplate | TaskTemplate[]): TemplateResponsePublic | TemplateResponsePublic[] {
+    if (Array.isArray(template)) {
+      return template.map((template) => {
+        return TemplateResponsePublicSchema.parse({
+          id: template.id,
+          object: 'taskTemplate',
+          name: template.title,
+          description: template.body,
+          createdDate: toRFC3339(template.createdAt),
+        })
+      })
+    }
```

Comment:
```suggestion
      return z.array(TemplateResponsePublicSchema).parse({
          id: template.id,
          object: 'taskTemplate',
          name: template.title,
          description: template.body,
          createdDate: toRFC3339(template.createdAt),
        })
      })
```
This we we run parse only one time, for 100 records this would run z.parse 100 times.